include/rocksdb/advanced_options.h

```diff
@@ -62,6 +62,13 @@ struct CompactionOptionsFIFO {
   // Default: 1GB
   uint64_t max_table_files_size;
 
+  // Drop files older than TTL. TTL based deletion will take precedence over
```

Comment:
You are right, and I too noticed it. I changed it now to fall back to size-based deletion when ttl-based deletion does not bring the total-size to be under the `max_table_files_size` threshold.

I am deferring the work of allow `max_table_files_size` to be set to 0 for now. I will look into it in future, as that involves a lot more changes.